### PR TITLE
[CDAP-13386] Fixes model training logs to open in logviewer

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Change the log viewer to show program logs.
- Modifies model training logs in MMDS to show logs in logviewer. Showing plaintext in UI breaks in authenticated environment.
- Minor refactor in terms of how we organize code in logviewer directive.

JIRA: https://issues.cask.co/browse/CDAP-13386
Build: https://builds.cask.co/browse/CDAP-UDUT39

### CODE REVIEW

File: cdap-ui/app/directives/log-viewer/log-viewer.js [hunk 1]
```diff
@@ -823,6 +728,77 @@ function LogViewerController ($scope, $window, LogViewerStore, myLogsApi, LOGVIE
       stopPoll();
     }
   });
+
+  if (vm.runId) {
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.js]: is this the start of the request? Perhaps add some comment to help indicate it? 😁

File: cdap-ui/app/logviewer/logviewer.less [hunk 2]
```diff
@@ -82,5 +82,8 @@ body.state-logviewerhome {
       height: ~"-moz-calc(100vh - 166px)";
       height: ~"-webkit-calc(100vh - 166px)";
     }
+    &.program-logs .logs-table {
+      height: ~"calc(100vh - 160px)";
+    }
```
Comment (elfenheart) [inline cdap-ui/app/logviewer/logviewer.less:87]: is the specificity of this change in `cdap-ui/app/directives/log-viewer/log-viewer.less` not enough?

**Comment by tonybach** [comment]
Talked to @ajainarayanan, currently if the split failed in the split step, we still show a link that shows logs in plain text.

Also, I'm seeing this for the 'End Time':

image

**Comment by elfenheart** [comment]
@ajainarayanan I think the issue that @tonybach brought up is a backend bug? if a split fail, shouldn't it still have an end time?

**Comment by ajainarayanan** [comment]
@elfenheart a split neither has a start time nor an end time. Need to file a JIRA for it.

**Comment by elfenheart** [comment]
@ajainarayanan in that case, what start time is displayed there in @tonybach screenshot? or is that not for a split?

**Comment by tonybach** [comment]
@elfenheart That's not the logs page for a failed split, that's when the model training step failed (which is after the split step). For the split step, we don't show the logs in logviewer yet:

split-failed-logs

**Comment by ajainarayanan** [comment]
@tonybach my latest changes should show the split logs as well. Are you not seeing it?

**Comment by ajainarayanan** [comment]
https://issues.cask.co/browse/CDAP-14100 Have filed this JIRA for adding start and end times for split and training info in an experiment. Will wait for the backend PR to merge before making changes to this PR.

File: cdap-ui/app/directives/log-viewer/log-viewer.js [hunk 3]
```diff
@@ -676,6 +579,7 @@ function LogViewerController ($scope, $window, LogViewerStore, myLogsApi, LOGVIE
   };
 
   vm.includeEvent = function(event, eventType) {
+    console.log('Entering include event');
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.js]: nit: remove

File: cdap-ui/app/directives/log-viewer/log-viewer.js [hunk 4]
```diff
@@ -712,8 +616,8 @@ function LogViewerController ($scope, $window, LogViewerStore, myLogsApi, LOGVIE
     // Whenever we change the log level filter, the data needs
     // to start from scratch
     vm.data = [];
-    vm.fromOffset = -10000 + '.' + vm.startTimeMs;
-
+    // And so does the offset!!
+    vm.fromOffset = null;
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.js]: is this alright? I think i recall some problem before if the `fromOffset` was null, that's why we did the `-10000.<some start time>`

Comment (ajainarayanan) [reply cdap-ui/app/directives/log-viewer/log-viewer.js]: @elfenheart The reason why this was changed was, when user changes the log level we used to use the start time from the last fetched log (line 168) and so the new polling mechanism doesn't retrieve any new logs. 
Talked with @poornachandra and @gsps1 regarding removing the fromoffset for the first request and the backend seems to be returning the latest 1 hour logs when no offset is mentioned. 

I can revert this back if you think this was causing some other side effect like UI getting less logs in last 1 hour and we no longer make any new request etc.,

**Review by elfenheart — APPROVED** [review]
👍 LGTM

File: cdap-ui/app/logviewer/logviewer.less (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/app/logviewer/logviewer.less:87]: Oh completely missed this comment. Yeah we apparently have css for logs table in logviewer directive and in the feature. Specifying the css for programs in logviewer directive was not sufficient.

**Pull request merged**